community/kernel/src/main/java/org/neo4j/kernel/impl/factory/GraphDatabaseFacadeFactory.java

```diff
@@ -0,0 +1,208 @@
+ * with custom versions that instantiate alternative services.
+ */
+public abstract class GraphDatabaseFacadeFactory
+{
+    public interface Dependencies
+    {
+        /**
+         * Allowed to be null. Null means that no external {@link org.neo4j.kernel.monitoring.Monitors} was created,
+         * let the
+         * database create its own monitors instance.
+         */
+        Monitors monitors();
+
+        LogProvider userLogProvider();
+
+        Iterable<Class<?>> settingsClasses();
+
+        Iterable<KernelExtensionFactory<?>> kernelExtensions();
+
+        Iterable<QueryEngineProvider> executionEngines();
+    }
+
+    public static class Configuration
+    {
+        public static final Setting<Boolean> read_only = GraphDatabaseSettings.read_only;
+        public static final Setting<Boolean> execution_guard_enabled = GraphDatabaseSettings.execution_guard_enabled;
+        public static final Setting<Boolean> ephemeral = setting( "ephemeral", Settings.BOOLEAN, Settings.FALSE );
+        public static final Setting<String> ephemeral_keep_logical_logs = setting("keep_logical_logs", STRING, "1 files", illegalValueMessage( "must be `true`/`false` or of format '<number><optional unit> <type>' for example `100M size` for " +
+                            "limiting logical log space on disk to 100Mb," +
+                            " or `200k txs` for limiting the number of transactions to keep to 200 000", matches(ANY)));
+        public static final Setting<File> store_dir = GraphDatabaseSettings.store_dir;
+        public static final Setting<File> neo_store = GraphDatabaseSettings.neo_store;
+
+        // Kept here to have it not be publicly documented.
+        public static final Setting<String> lock_manager = setting( "lock_manager", Settings.STRING, "" );
+        public static final Setting<String> tracer =
+                setting( "dbms.tracer", Settings.STRING, (String) null ); // 'null' default.
+
+        public static final Setting<String> editionName = setting("edition", Settings.STRING, "Community");
+    }
+
+    /**
+     * Instantiate a graph database given configuration and dependencies.
+     *
+     * @param params
+     * @param dependencies
+     * @return
+     */
+    public GraphDatabaseFacade newFacade( Map<String, String> params, final Dependencies dependencies )
+    {
+        return newFacade( params, dependencies, new GraphDatabaseFacade() );
+    }
+
+    /**
+     * Instantiate a graph database given configuration, dependencies, and a custom implementation of {@link org.neo4j.kernel.impl.factory.GraphDatabaseFacade}.
+     *
+     * @param params
+     * @param dependencies
+     * @param graphDatabaseFacade
+     * @return
+     */
+    public GraphDatabaseFacade newFacade( Map<String, String> params, final Dependencies dependencies,
+                                          final GraphDatabaseFacade graphDatabaseFacade )
```

Comment:
Is this meant to be private? (Factory methods don't take already constructed objects of the type they produce)